packages/apps-sdk/src/sdk.ts

```diff
@@ -31,9 +34,21 @@ export class SDK {
    */
   private configureApp(): void {
     // First we obtaing the id of this app from the url
-    const parsed = queryString.parse(window.location.search);
+    const parsed = qs.parse(window.location.search.substring(1), { decoder });
     const parsedAppName = Array.isArray(parsed.appName) ? parsed.appName[0] : parsed.appName;
+    const parsedQueryParams = Array.isArray(parsed.queryParams) ? parsed.queryParams[0] : parsed.queryParams;
+    const parsedInitialData = Array.isArray(parsed.initialData) ? parsed.initialData[0] : parsed.initialData;
+
     this.appName = parsedAppName ? parsedAppName : 'messenger-iframe';
+    this.queryParams = parsedQueryParams;
+    this.initialData = parsedInitialData;
+
+    const mode = this.queryParams && this.queryParams.mode ? this.queryParams.mode : APP_MODE_IFRAME;
+    const channgelConfig = mode === APP_MODE_POPUP ? { window: window.opener } : { window: window.parent };
+
+    // Define Communication Channels
+    this.channelManager = this.channelFactory();
```

Comment:
What is the difference between `channelManager` that you assigns here and `channelFactory` that you assigned at constructor? Why not to assign manager also on init ?

Reply:
channelFactory at the constructor is a factory that create and object of the type channelManager 
The intention of the channelFactory is to allow us to change for another factory in the future (still need to Type better its desired interface) and mock this dependency so it make easier to test this clase without having to do real postMessage (post-robot) data exchange.

channelManager is the instance of the factory that allows us to call different methods to communicate (i.e send), this could be an instance of a socket library in the future